### PR TITLE
[#4479] Add irepl path_with_no_perms test (master)

### PR DESCRIPTION
**Pull request by alanking**
The test passes!

### CODE REVIEW

**Comment by trel** [comment]
the test reproduces the issue?    or proves the issue is now invalid because it's already been fixed in 4-2-stable?

**Comment by alanking** [comment]
The test proves that the issue is now invalid because it has already been fixed upstream (both master and 4-2-stable). Adjust tests to use `iscan` to ensure files appear on disk.

**Comment by alanking** [comment]
#'d

File: scripts/irods/test/test_irepl.py [hunk 1]
```diff
@@ -163,3 +163,55 @@ def test_irepl_not_honoring_R_option__issue_3885(self):
...
+                'STDOUT', 'unixfilesystem')
+            self.admin.assert_icommand(
+                ['iadmin', 'mkresc', resource_2, 'unixfilesystem', ':'.join([test.settings.HOSTNAME_2, resource_2_vault])],
+                'STDOUT', 'unixfilesystem')
+
+            filename = 'test_irepl_data_object_with_no_permission__4479'
+            physical_path = os.path.join(user0.local_session_dir, filename)
+            lib.make_file(physical_path, 1024)
+            logical_path = os.path.join(user0.session_collection, filename)
+            logical_path_sans_zone = os.sep.join(str(logical_path).split(os.sep)[2:])
```
Comment (trel) [inline scripts/irods/test/test_irepl.py]: Codacy Issue found: [Unused variable 'logical_path_sans_zone'](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5120499)

**Pull request merged**